bundles/net.tourbook.device/src/net/tourbook/device/hac4linux/tour/HAC4LinuxDeviceReader.java

```diff
@@ -45,667 +45,701 @@
+
+            final TimeData firstTimeData = timeDataList.get(0);
+
+            if (tourData.getDeviceName().equals("CM414AM")) { //$NON-NLS-1$
+               firstTimeData.pulse = Float.MIN_VALUE;
+            }
+         }
+
+         tourData.createTimeSeries(timeDataList, true);
+
+         // after all data are added, the tour id can be created
+         final int tourDistance = (int) Math.abs(tourData.getStartDistance());
+         final String uniqueId = createUniqueId_Legacy(tourData, tourDistance);
+         final Long tourId = tourData.createTourId(uniqueId);
+
+         // check if the tour is in the tour map
+         if (alreadyImportedTours.containsKey(tourId) == false) {
+
+            // add new tour to the map
+            newlyImportedTours.put(tourId, tourData);
+
+            // create additional data
+            tourData.computeComputedValues();
+            tourData.computeTourDrivingTime();
+
+            tourData.completeTourMarkerWithRelativeTime();
+         }
+
+      } catch (final Exception e) {
+         e.printStackTrace();
+         return false;
+      } finally {
+         if (fileHac4LinuxData != null) {
+            try {
+               fileHac4LinuxData.close();
+            } catch (final IOException e1) {
+               e1.printStackTrace();
+            }
+         }
+      }
+
+      return true;
+
+   }
+
+   /*
+    * (non-Javadoc) Switching the machine to another state depending on the section of the file
+    * currently read.
+    */
+   private void switchSection(final String line) {
+      if (line.equals("[FILE]")) {
+//$NON-NLS-0$
+         m_section = Section.SECTION_FILE;
+         return;
+      }
+      if (line.equals("[INFORMATION]")) {
+//$NON-NLS-0$
+         m_section = Section.SECTION_INFO;
+         return;
+      }
+      if (line.equals("[NOTES]")) {
+//$NON-NLS-0$
+         m_section = Section.SECTION_NOTES;
+         return;
+      }
+      if (line.equals("[FRIENDS]")) {
+//$NON-NLS-0$
+         m_section = Section.SECTION_FRIENDS;
+         return;
+      }
+      if (line.equals("[PERSON]")) {
+//$NON-NLS-0$
+         m_section = Section.SECTION_PERSON;
+         return;
+      }
+      if (line.equals("[STATISTICS]")) {
+//$NON-NLS-0$
+         m_section = Section.SECTION_STATISTICS;
+         return;
+      }
+      if (line.equals("[SETTINGS]")) {
+//$NON-NLS-0$
+         m_section = Section.SECTION_SETTINGS;
+         return;
+      }
+      if (line.equals("[POLAR-EXTENSION]")) {
+//$NON-NLS-0$
+         m_section = Section.SECTION_POLAREXTS;
+         return;
+      }
+      if (line.equals("[COACH-PARAMETER]")) {
+//$NON-NLS-0$
+         m_section = Section.SECTION_COACH;
+         return;
+      }
+      if (line.equals("[TOUR-DATA]")) {
+//$NON-NLS-0$
+         m_section = Section.SECTION_TOURDATA;
+         return;
+      }
+      if (line.equals("[MARKING-DATA]")) {
+//$NON-NLS-0$
+         m_section = Section.SECTION_MARKS;
+         return;
+      }

```

Comment:
This formatting looks very awful compared with the previous formatting. It would have been better to use the formatting off/on tags for manual formatting

Reply:
Sometimes Eclipse does weird thing with the formatting. I will revert to the original formatting if you prefer

Reply:
Fixed